### PR TITLE
cherry pick dynamo library from subscriptions branch

### PR DESCRIPTION
**Pull request by Jlrine2**

### CODE REVIEW

File: tests/api/conftest.py [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-from moto import mock_dynamodb2
 
-from hyp3_api import CMR_URL, DYNAMODB_RESOURCE, app, auth
+from hyp3_api import CMR_URL, app, auth
 from hyp3_api.util import get_granules
 
 AUTH_COOKIE = 'asf-urs'
 JOBS_URI = '/jobs'
 USER_URI = '/user'
+SUBSCRIPTIONS_URI = '/subscriptions'
```
Comment (asjohnston-asf) [inline tests/api/conftest.py:13]: can drop from the cherry-pick for now

File: lib/dynamo/dynamo/jobs.py [hunk 2]
```diff
@@ -2,23 +2,8 @@
...
-    if formatted_start and formatted_end:
-        return key.between(formatted_start, formatted_end)
-    if formatted_start:
-        return key.gte(formatted_start)
-    if formatted_end:
-        return key.lte(formatted_end)
+from dynamo.util import DYNAMODB_RESOURCE, get_request_time_expression
 
 
 def put_jobs(payload: List[dict]):
```
Comment (asjohnston-asf) [inline lib/dynamo/dynamo/jobs.py:9]: I think we should move the business logic to add the request_time, status_code, and job_id fields into this function, so we don't have to duplicate it between the `api` and `process_new_granules`. That might move a few additional functions from api.util to dynamo.util.

File: tests/api/conftest.py (hunk 1 above)
Comment (Jlrine2) [reply tests/api/conftest.py:13]:
```suggestion
```

File: lib/dynamo/dynamo/jobs.py (hunk 2 above)
Comment (asjohnston-asf) [reply lib/dynamo/dynamo/jobs.py:9]: took a stab at this in https://github.com/ASFHyP3/hyp3/pull/512 , which is ready for review

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.1.0](https://github.com/ASFHyP3/hyp3/compare/v2.0.4...v2.1.0)
+### Added
+- `lib/dynamo` library to allow sharing common code among different apps.
+
+## Changed
+- moved dynamodb functinoality from `hyp3_api/dynamo` to `lib/dynamo`
```
Comment (asjohnston-asf) [inline CHANGELOG.md:12]:
```suggestion
- moved dynamodb functionality from `hyp3_api/dynamo` to `lib/dynamo`
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.1.0](https://github.com/ASFHyP3/hyp3/compare/v2.0.4...v2.1.0)
+### Added
+- `lib/dynamo` library to allow sharing common code among different apps.
+
+## Changed
```
Comment (asjohnston-asf) [inline CHANGELOG.md:11]:
```suggestion
## Changed
- `POST /jobs` responses no longer include the `job_id`, `request_time`, `status_code`, or `user_id` fields when `validate_only=true` 
```

**Review by asjohnston-asf — APPROVED** [review]

**Pull request merged**